includes/template-library/manager.php

```diff
@@ -766,10 +777,32 @@ private function is_allowed_to_read_template( array $args ): bool {
 			$this->set_wordpress_adapter( new WordPress_Adapter() );
 		}
 
+		if ( ! $this->should_check_permissions( $args ) ) {
+			return true;
+		}
+
 		$post_id = intval( $args['template_id'] );
 		$post_status = $this->wordpress_adapter->get_post_status( $post_id );
 		$is_private_or_non_published = ( 'private' === $post_status && ! $this->wordpress_adapter->current_user_can( 'read_private_posts', $post_id ) ) || ( 'publish' !== $post_status );
 
-		return $is_private_or_non_published || $this->wordpress_adapter->current_user_can( 'edit_post', $post_id );
+		$can_edit_template = $is_private_or_non_published || $this->wordpress_adapter->current_user_can( 'edit_post', $post_id );
+
+		return apply_filters( 'elementor/template-library/is_allowed_to_read_template', $can_edit_template, $args );
+	}
+
+	private function should_check_permissions( array $args ): bool {
+		if ( null === $this->elementor_adapter ) {
+			$this->set_elementor_adapter( new Elementor_Adapter() );
+		}
+
+		if( ! isset( $args['check_permissions'] ) ) {
```

Comment:
```suggestion
		if ( ! isset( $args['check_permissions'] ) ) {
```
Lint will probably fail